### DISCUSSION TITLE
TG Station Setup

### DISCUSSION THREAD

**Discussion by antt1995** [tgstation/tgstation-server-tools#1846] [Q&A] [answered]
Hi,
I have Installed this on my Windows server with a fresh install of MariaDB,
I have setup the Repo, that has worked to Download TGStation git,
When I try and Deploy the server it keeps failing as per the Image Below:
![image](https://github.com/user-attachments/assets/804e7dee-34ef-4217-bf45-ad85fdc30488)

I can Manually run the build.bat with no Issues

Any Suggestions?

**Comment by antt1995** [comment]
![image](https://github.com/user-attachments/assets/5799c8b8-fa16-45fb-a0a4-b15e7606de8c)

**Reply by Cyberboss** [reply]
Please check/upload your TGS logs that contain a run of the deployment.

If you can run it locally fine and it's timing out in TGS after an hour it likely means something is causing it to deadlock for all that time. The logs will show what exactly

**Reply by antt1995** [reply]
[tgs-20240801.log](https://github.com/user-attachments/files/16457358/tgs-20240801.log)
As Requested

**Reply by antt1995** [reply]
I have had a look and the main thing I notice is that its Trying to run from the D Drive when its all Installed and Running from the C Drive

**Reply by Cyberboss** [reply]
No, that's the TGS callstack (it's built on the D: drive).

What's happening is your precompile script (`C:\Temp\SS13\Configuration\EventScripts\PreCompile.bat`) is timing out. Try debugging that?

**Reply by antt1995** [reply]
I have not made this, is it from the tgstation repo

[PreCompile.txt](https://github.com/user-attachments/files/16458858/PreCompile.txt)

**Reply by Cyberboss** [reply]
Do you know how to use process explorer? If you start the deployment and wait 10 minutes what does the Tgstation.Server.Host.Service process tree look like?

**Reply by antt1995** [reply]
![image](https://github.com/user-attachments/assets/e76a90ef-2610-47ff-adb5-90f0034bf116)
![image](https://github.com/user-attachments/assets/e3cb27bf-1ca2-4fb7-bbd2-d4640506eb6f)
CMD: ``C:\Windows\system32\cmd.exe /c ""C:\Temp\SS13\Configuration\EventScripts\PreCompile.bat" C:\Temp\SS13\Game\882bfd26-89d0-47fa-8007-1ea8ced5dae2 https://github.com/tgstation/tgstation OpenDream-e6b0e49aa007b72a062837bb8e00b45f224ce906 master"``
DIR: ``C:\Temp\SS13\Game\882bfd26-89d0-47fa-8007-1ea8ced5dae2\``

**Reply by Cyberboss** [reply]
What happens if you try to run that command line manually?

As a prerequisite, copy your Repository folder to the directory shown in the command line.

**Reply by antt1995** [reply]
![image](https://github.com/user-attachments/assets/13beacb4-4b33-4107-ad60-3f5305f07cc5)
``C:\Windows\system32\cmd.exe /c ""C:\Temp\SS13\Configuration\EventScripts\PreCompile.bat" C:\Temp\SS13\Game\882bfd26-89d0-47fa-8007-1ea8ced5dae2 https://github.com/tgstation/tgstation OpenDream-e6b0e49aa007b72a062837bb8e00b45f224ce906 master"
Using vendored Node Access is denied.
Access is denied.``

**Reply by antt1995** [reply]
Running Windows Server Datacentre 2022,

Get the same message when trying to launch node.exe manually from
C:\Temp\SS13\Configuration\EventScripts\node-v20.13.0-x64

**Reply by Cyberboss** [reply]
Ah, sorry I missed that. If you can figure out why node.exe isn't working you'll be set.

The binary _may_ be corrupt, so possibly try deleting it and letting it redownload.

**Reply by antt1995** [reply, marked as answer]
So Deleting the Node Folder and Letting it Use the Node that's installed seems to have worked,
![image](https://github.com/user-attachments/assets/99b8798e-0b72-48fe-8425-92f207178338)

**Reply by Cyberboss** [reply]
The Precompile.bat script downloads a fresh node if it is missing. It looks like your first download Wass corrupt for some reason (node is notorious for this in my experience).
